aexpect/ops_linux.py

```diff
@@ -216,6 +216,29 @@ def is_nonzero_size_file(session, path):
 ###############################################################################
 
 
+def glob(session, glob_pattern):
+    """
+    Find files matching given pattern through a session.
+
+    :param session: session to run the command on
+    :type session: ShellSession
+    :param str glob_pattern: pattern of filenames where `*`, `?` and such match
+                             any string or not-empty-string and so on.
+    :returns: file names matching pattern, including path (like glob.glob)
+    :rtype: [str]
+
+    Much like :py:func:`glob.glob` but implemented using `echo`.
+    """
+    cmd = f'echo {glob_pattern}'
+    status, output = session.cmd_status_output(cmd)
+    status, output = _process_status_output(cmd, status, output)
+    paths = []
+    for path in output.split():
```

Comment:
This will not work for paths containing spaces. (eg `touch "a b"`)

Reply:
Hmmm this is indeed a limitation however, I would prefer to simply put a disclaimer in the docstring as there is no better solution I would prefer. As you can see in our original discussion the only other option is shell specific and I would prefer not to go in this direction. We already ruled out the `ls` hiding status and this is the only option that remains. Is this ok?

Reply:
I'd probably use `MATCHES=({globpattern})\nfor MATCH in "${MATCHES[@]}"; do echo "$MATCH"; done` although I'm not 100% sure we won't end up with some weird chars issues as well.

Reply:
This sounds pretty much like my comment https://github.com/avocado-framework/aexpect/pull/98#discussion_r988658483 which remains shell specific or am I missing something?